### PR TITLE
Cover Category CommandHandlers and QueryHandlers by behat tests

### PR DESCRIPTION
**Pull request by tdavidsonas88**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | increases behat tests coverage for Category Management feature related handlers
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/14480
| How to test?  | CI build should pass

GetCategoryForEditing
EditCategory
DeleteCategory
AddCategory refactored to use domain
BulkDeleteCategories
UpdateCategoryPosition
EditRootCategory

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16966)

### CODE REVIEW

File: src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php [hunk 1]
```diff
@@ -151,48 +151,53 @@ private function getThumbnailImage(CategoryId $categoryId)
...
-                        true
-                    );
+        if (count($imageTypes) > 0) {
+            $thumb = '';
+            $imageTag = '';
+            $formattedSmall = ImageType::getFormattedName('small');
+            $imageType = new ImageType();
+            foreach ($imageTypes as $k => $imageType) {
+                if ($formattedSmall == $imageType['name']) {
+                    $thumb = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '-' . $imageType['name'] . '.jpg';
```
Comment (matks) [inline src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php:161]: I see you stumbled upon this code we are not proud of 😄.
Unfortunately we kept the previous behavior which is a bit raw, but we might be able to improve it in the future.

File: tests/Integration/Behaviour/Features/Context/Util/CategoryTreeIterator.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Tests\Integration\Behaviour\Features\Context\Util;
+
+use PrestaShop\PrestaShop\Adapter\Form\ChoiceProvider\CategoryTreeChoiceProvider;
+
+class CategoryTreeIterator
```
Comment (matks) [inline tests/Integration/Behaviour/Features/Context/Util/CategoryTreeIterator.php:7]: Nice way to handle this obstacle 👍

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @tdavidsonas88

File: src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php (hunk 1 above)
Comment (tdavidsonas88) [reply src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php:161]: Thanks for pointing this out, resolved this stumbling in a different way in Behat tests in the next pull request.

**Pull request merged**